packages/svelte/tests/runtime-runes/samples/effect-cleanup/_config.js

```diff
@@ -0,0 +1,19 @@
+import { test } from '../../test';
+import { flushSync } from 'svelte';
+
+export default test({
+	get props() {
+		return { log: [] };
+	},
+
+	async test({ assert, target, component }) {
+		const [b1] = target.querySelectorAll('button');
+		flushSync(() => {
+			b1.click();
+		});
+		flushSync(() => {
+			b1.click();
+		});
+		assert.deepEqual(component.log, ['init 0', 'cleanup 2', 'init 2', 'cleanup 4', 'init 4']);
```

Comment:
Shouldn't this be?

```
		assert.deepEqual(component.log, ['init 0', 'cleanup 0', 'init 2', 'cleanup 2', 'init 4']);
```

such that the clean up of the previous effect runs before the init of the new one. That'd be the expected behaviour to me.

Reply:
The values update synchronously, but effects trigger async. It's expected the value should reflect the latest.

Reply:
If it was a normal value in the closure instead of a `derived` would that work as expected?

Reply:
Not sure I follow? It would be the same in both cases, no?

Reply:
Yes it would, however derived is a computed function so you can imagine it more like calling the function to get the double value again in the cleanup.